tests/test_regex.py

```diff
@@ -43,6 +43,10 @@ def test_helper_validate_invalid(self) -> None:
         """Should pass validation for valid regular expression"""
         self.assertFalse(re._validate("a(|)"))
 
+    def test_validate_unicode_characters(self) -> None:
+        """Should pass validation for regular expressions with unicode characters."""
+        re.validate("(µ|🤖ù)*")
+
```

Comment:
Should add a test that an nfa converted from this regex has the expected set of input symbols.

Reply:
Hello,

I added a test for this (not sure if it it should be in `test_regex.py` or `test_nfa.py`, so I put it in the former).

I also changed a bit of `NFA.from_regex(regex, input_symbols)` (which could be reverted, or be its own issue).
Currently, the default behaviour for when input_symbols is None is to default to the set of ascii lowercase letters and numbers.
Wouldn't it make more sense if it rather defaulted to all letters present in the regex ?

Reply:
I think `test_regex` is a fine location to put this test.

For the second item, the prior default behavior made sense when the allowable alphabet was small, but now allowing all unicode characters, I think your choice of a default makes sense 👍🏽